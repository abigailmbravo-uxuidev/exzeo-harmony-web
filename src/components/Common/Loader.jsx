import React from 'react';
import propTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import * as appStateActions from '../../actions/appStateActions';

const Loader = props => (
  <div className="loader modal">
    <div className="card" role="dialog">
      <div className="card-block">
        <div className="ring-5">
          <div className="ring-4">
            <div className="ring-3">
              <div className="ring-2">
                <div className="ring-1" />
              </div>
            </div>
          </div>
        </div>
        <span>Loading</span>
      </div>
    </div>
  </div>
);

Loader.propTypes = {
  appState: propTypes.shape({
    data: propTypes.shape({
      nextPage: propTypes.string
    })
  })
};

const mapStateToProps = state => ({
  tasks: state.cg,
  appState: state.appState
});

const mapDispatchToProps = dispatch => ({
  actions: {
    appStateActions: bindActionCreators(appStateActions, dispatch)
  }
});

// ------------------------------------------------
// wire up redux form with the redux connect
// ------------------------------------------------
export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({ form: 'Loader' })(Loader));

// export default Loader;
