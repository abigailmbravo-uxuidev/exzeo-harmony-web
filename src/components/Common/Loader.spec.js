import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

it('renders without crashing', () => {
  const tree = renderer.create(
    <Loader />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
