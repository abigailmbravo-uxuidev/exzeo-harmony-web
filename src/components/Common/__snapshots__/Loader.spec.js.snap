exports[`test renders without crashing 1`] = `
<div
  className="loader modal">
  <div
    className="card"
    role="dialog">
    <div
      className="card-block">
      <div
        className="ring-5">
        <div
          className="ring-4">
          <div
            className="ring-3">
            <div
              className="ring-2">
              <div
                className="ring-1" />
            </div>
          </div>
        </div>
      </div>
      <span>
        Loading
      </span>
    </div>
  </div>
</div>
`;
